docs/adr/ADR-023-sequencer-recovery.md

```diff
@@ -0,0 +1,132 @@
+- `StopSequencer()` — hard stop; no more block production.
+- `SequencerHealthy()` → `{ healthy, l2_number, l2_hash, l1_origin, peer_count, da_height, last_err }`
+- `Status()` → `{ sequencer_active, build_height, leader_hint?, last_err }`
+
+These are additive and should not break existing RPCs.
+
+### Efficiency considerations
+- **Design A:** Raft heartbeats and snapshotting add small steady‑state overhead; no impact on throughput when healthy.
+- **Design B:** Lease renewals are lightweight; performance dominated by `ev-node` itself.
+
+### Expected access patterns
+- Reads (RPC, state) should work on all nodes; **writes/sequence endpoints** only on the active leader. If a leader‑aware proxy is deployed, it enforces this automatically.
+
+### Logging/Monitoring/Observability
+- Metrics: `leader_id`, `raft_term` (A), `lease_owner` (B), `unsafe_head_advance`, `peer_count`, `rpc_error_rate`, `da_publish_latency`, `backlog`.
+- Alerts: no unsafe advance > 3× block time; unexpected leader churn; lease lost but sequencer still active (fencing breach).
+- Logs: audit all **Start/Stop** decisions and override operations.
+
+### Security considerations
+- Lock down **Admin RPC** with mTLS + RBAC; only the sidecar/process account may call Start/Stop.
+- Implement **fencing**: leader periodically validates it still holds leadership/lease; otherwise self‑stops.
+- Break‑glass overrides must be gated behind separate credentials and produce auditable events.
+
+### Privacy considerations
+- None beyond existing node telemetry; no user data added.
+
+### Testing plan
+- Kill active sequencer → verify failover within SLO; assert **no double leadership**.
+- Partition tests: only Raft majority (A) or lease holder (B) may produce.
+- Blue/green: explicit leadership handoff; confirm unsafe head continuity.
+- Misconfigured standby → failover should **refuse**; alarms fire.
+- Long‑duration outage drills; confirm user‑facing status and catch‑up behavior.
+
+### Change breakdown
+- Phase 1: Implement Admin RPC + health surface in `ev-node`; add sidecar skeletons.
+- Phase 2: Integrate Design A (Raft) in a 1 sequencer + 2 failover; build dashboards/runbooks.
+- Phase 3: Add Design B (Lease) profile for small/test clusters; share common health logic.
+- Phase 4: Game days and SLO validation; finalize SRE playbooks.
+
+### Release/compatibility
+- **Breaking release?** No — Admin RPCs are additive.
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+- Clear, deterministic leadership with fencing; supports zero‑downtime upgrades.
+- Works with `ev-node` via a small, well‑defined Admin RPC.
+- Choice of control plane allows right‑sizing ops: Raft for prod; Lease for small/test.
+
+### Negative
+- Design A adds Raft operational overhead (quorum management, snapshots).
+- Design B has a smaller blast radius but does not generalize to N replicas; stricter reliance on correct fencing.
```

Comment:
Depending on the chosen implementation, the sequencer stack still may possess a single point of failure (i.e kv store)

Reply:
the failure being the other node is not up to date with the latest state?

Reply:
the kv store will always be local to the node, we dont support adding remote kv stores (dbs)